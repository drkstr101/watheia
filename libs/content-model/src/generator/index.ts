export { default } from './type-generator';
