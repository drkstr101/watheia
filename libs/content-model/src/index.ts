// export core types
export type { ConfigModel, DataModel, Model, ObjectModel, PageModel } from '@stackbit/types';

// model exports
export { ALL_MODEL_NAMES, models } from './models';
export type { ContentModel, ModelName } from './models';

export { types } from './types';
export type {
  DataModelProps,
  DocumentModelProps as DocumentProps,
  ModelProps,
  ObjectModelProps,
  PageModelProps as PageProps,
} from './types';
