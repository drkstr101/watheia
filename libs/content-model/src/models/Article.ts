import { PageModel } from '@stackbit/types';

export const Article: PageModel = {
  type: 'page',
  name: 'Article',
  label: 'Article',
  urlPath: '/blog/{slug}',
  filePath: 'content/pages/blog/{slug}.md',
  hideContent: true,
  thumbnail: 'https://assets.stackbit.com/components/models/thumbnails/default.png',
  fields: [
    {
      type: 'string',
      name: 'slug',
      label: 'Slug',
      description: 'A unique url for this page starting with "/"',
      required: true,
    },
    {
      type: 'string',
      name: 'title',
      label: 'Title',
      default: 'This Is a New Page',
      required: true,
    },
    {
      type: 'string',
      name: 'description',
      label: 'Description',
      default: 'A brief (1-3 sentence) summary of the content',
      required: true,
    },
    {
      type: 'date',
      name: 'date',
      label: 'The listed publication date',
      required: true,
    },
    {
      type: 'reference',
      name: 'author',
      label: 'Author',
      models: ['Person'],
    },
  ],
};
