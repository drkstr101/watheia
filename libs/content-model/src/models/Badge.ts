import { Model } from '@stackbit/types';

export const Badge: Model = {
  type: 'object',
  name: 'Badge',
  label: 'Badge',
  labelField: 'label',
  fields: [
    {
      type: 'string',
      name: 'label',
      label: 'Label',
      description: 'The text in the badge',
      required: true,
      default: 'This is a badge',
      hidden: false,
      localized: false,
    },
    {
      type: 'enum',
      name: 'color',
      label: 'Color',
      required: false,
      default: 'primary',
      hidden: false,
      localized: false,
      options: [
        {
          label: 'Dark',
          value: 'black',
          textColor: '$colors.black',
          backgroundColor: '$colors.black',
          borderColor: '#ececec',
        },
        {
          label: 'Light',
          value: 'white',
          textColor: '$colors.white',
          backgroundColor: '$colors.white',
          borderColor: '#ececec',
        },
        {
          label: 'Neutral',
          value: 'neutral',
          textColor: '$colors.neutral.200',
          backgroundColor: '$colors.neutral.200',
          borderColor: '#ececec',
        },
        {
          label: 'Primary',
          value: 'primary',
          textColor: '$colors.primary.500',
          backgroundColor: '$colors.primary.500',
          borderColor: '#ececec',
        },
      ],
      group: 'styles',
      controlType: 'palette',
    },
    {
      type: 'style',
      name: 'styles',
      label: 'Styles',
      description: 'The styles field is controlled by Stackbit editor',
      required: false,
      hidden: false,
      localized: false,
      styles: {
        self: {
          fontStyle: '*',
          fontWeight: ['400', '500', '700'],
          textDecoration: '*',
          textAlign: '*',
        },
      },
    },
  ],
  fieldGroups: [
    {
      name: 'styles',
      label: 'Styles',
      icon: 'palette',
    },
  ],
};
