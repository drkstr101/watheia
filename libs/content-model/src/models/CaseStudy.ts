import { PageModel } from '@stackbit/types';

export const CaseStudy: PageModel = {
  type: 'page',
  name: 'CaseStudy',
  label: 'Page',
  urlPath: '/work/{slug}',
  filePath: 'content/pages/work/{slug}.md',
  hideContent: true,
  thumbnail: 'https://assets.stackbit.com/components/models/thumbnails/default.png',
  fields: [
    {
      type: 'string',
      name: 'slug',
      label: 'Slug',
      description: 'A unique url for this page starting with "/"',
      required: true,
    },
    {
      type: 'string',
      name: 'title',
      label: 'Title',
      default: 'This Is a New Page',
      required: true,
    },
    {
      type: 'string',
      name: 'description',
      label: 'Description',
      default: 'A brief (1-3 sentence) summary of the content',
      required: true,
    },
    {
      type: 'date',
      name: 'date',
      label: 'The publication date',
      required: true,
    },
    {
      type: 'string',
      name: 'client',
      label: 'Client name',
      default: 'Acme',
      required: true,
    },
    {
      type: 'list',
      name: 'summary',
      label: 'Summary',
      description: 'A list of paragraphs used for the summary',
      required: true,
      items: { type: 'text' },
    },
    {
      type: 'image',
      name: 'logo',
      label: 'Client logo',
      default: '/images/default.png',
      required: true,
    },
    {
      type: 'model',
      name: 'image',
      label: 'Image',
      models: ['Image'],
      required: true,
    },
    {
      type: 'list',
      name: 'services',
      label: 'Service tags',
      description: 'A list of services used in this case study',
      required: true,
      items: { type: 'text' },
    },
    {
      type: 'model',
      name: 'testimonial',
      label: 'Testimonial',
      models: ['Testimonial'],
      required: true,
    },
  ],
};
