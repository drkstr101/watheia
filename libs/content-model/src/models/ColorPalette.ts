import { ObjectModel } from '@stackbit/types';

export const ColorPalette: ObjectModel = {
  type: 'object',
  name: 'ColorPalette',
  label: 'Color palette',
  fields: [
    {
      type: 'string',
      name: '100',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '200',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '300',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '400',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '500',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '600',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '700',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '800',
      required: true,
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: '900',
      required: true,
      hidden: false,
      localized: false,
    },
  ],
  fieldGroups: [],
};
