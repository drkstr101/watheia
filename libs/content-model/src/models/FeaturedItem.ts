import { ObjectModel } from '@stackbit/types';

export const FeaturedItem: ObjectModel = {
  type: 'object',
  name: 'FeaturedItem',
  label: 'Item',
  labelField: 'title',
  fields: [
    {
      type: 'string',
      name: 'title',
      label: 'Title',
      required: true,
      default: 'This Is The Item Title',
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: 'tagline',
      label: 'Tagline',
      description:
        'A small short text above the title. Also used as a tab title in the tab navigation arrangement of carousel section.',
      required: false,
      default: 'This is the tagline',
      hidden: false,
      localized: false,
    },
    {
      type: 'string',
      name: 'subtitle',
      label: 'Subtitle',
      required: false,
      default: 'This is the item subtitle',
      hidden: false,
      localized: false,
    },
    {
      type: 'markdown',
      name: 'text',
      label: 'Text',
      required: false,
      default:
        'Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae. explicabo.',
      hidden: false,
      localized: false,
    },
    {
      type: 'model',
      name: 'image',
      label: 'Image',
      required: false,
      hidden: false,
      localized: false,
      models: ['ImageBlock'],
    },
    {
      type: 'list',
      name: 'actions',
      label: 'Actions',
      required: false,
      hidden: false,
      localized: false,
      items: {
        type: 'model',
        models: ['Button', 'Link'],
      },
    },
    {
      type: 'string',
      name: 'elementId',
      label: 'Element ID',
      description: 'The unique ID for an HTML element, must not contain whitespace',
      required: false,
      hidden: false,
      localized: false,
      group: 'settings',
    },
    {
      type: 'enum',
      name: 'colors',
      label: 'Colors',
      required: false,
      default: 'bg-light-fg-dark',
      hidden: false,
      localized: false,
      options: [
        {
          label: 'Light background, dark foreground',
          value: 'bg-light-fg-dark',
          textColor: '$colors.black',
          backgroundColor: '$colors.white',
          borderColor: '#ececec',
        },
        {
          label: 'Neutral background, dark foreground',
          value: 'bg-neutral-fg-dark',
          textColor: '$colors.black',
          backgroundColor: '$colors.neutral.200',
          borderColor: '#ececec',
        },
        {
          label: 'Neutral alt background, dark foreground',
          value: 'bg-neutralAlt-fg-dark',
          textColor: '$colors.black',
          backgroundColor: '$colors.neutral.300',
          borderColor: '#ececec',
        },
        {
          label: 'Dark background, light foreground',
          value: 'bg-dark-fg-light',
          textColor: '$colors.white',
          backgroundColor: '$colors.black',
          borderColor: '#ececec',
        },
      ],
      group: 'cardStyles',
      controlType: 'palette',
    },
    {
      type: 'style',
      name: 'styles',
      label: 'Styles',
      description: 'The styles field is controlled by Stackbit editor',
      required: false,
      hidden: false,
      localized: false,
      styles: {
        self: {
          margin: ['tw0:96'],
          padding: ['tw0:96'],
          flexDirection: '*',
          justifyContent: ['flex-start', 'flex-end', 'center'],
          borderWidth: ['0:2', '4:8:4'],
          borderStyle: '*',
          borderColor: [
            {
              value: 'border-dark',
              label: 'Dark',
              color: '$colors.black',
            },
            {
              value: 'border-light',
              label: 'Light',
              color: '$colors.white',
            },
            {
              value: 'border-neutral',
              label: 'Neutral',
              color: '$colors.neutral.200',
            },
            {
              value: 'border-neutralAlt',
              label: 'Neutral alt',
              color: '$colors.neutral.300',
            },
            {
              value: 'border-primary',
              label: 'Primary',
              color: '$colors.primary.500',
            },
          ],
          borderRadius: '*',
          textAlign: '*',
        },
      },
    },
  ],
  fieldGroups: [
    {
      name: 'cardStyles',
      label: 'Card styles',
      icon: 'palette',
    },
    {
      name: 'settings',
      label: 'Settings',
      icon: 'gear',
    },
  ],
};
