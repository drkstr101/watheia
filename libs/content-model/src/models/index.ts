import { Article } from './Article';
import { Button } from './Button';
import { Card } from './Card';
import { CardsSection } from './CardsSection';
import { CaseStudy } from './CaseStudy';
import { Config } from './Config';
import { Footer } from './Footer';
import { Header } from './Header';
import { HeroSection } from './HeroSection';
import { Image } from './Image';
import { Link } from './Link';
import { Page } from './Page';
import { Person } from './Person';
import { Theme } from './Theme';

export const models = {
  Article,
  Button,
  Card,
  CardsSection,
  CaseStudy,
  Config,
  Footer,
  Header,
  HeroSection,
  Image,
  Link,
  Page,
  Person,
  Theme,
} as const;
