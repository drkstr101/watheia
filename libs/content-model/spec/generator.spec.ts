import generateModelTypes from '../src/generator';
import { models } from '../src/models';

describe('watheia.cabbage/type-generator', () => {
  // afterEach(() => rmSync(tempFile));
  describe('generateModelTypes', () => {
    it('generates an empty file when no models provided', () => {
      const output = generateModelTypes([]);
      expect(output).toBeEmpty();
    });

    it('transforms simple model types', () => {
      const output = generateModelTypes([
        {
          name: 'Page',
          type: 'page',
          fields: [
            { name: 'slug', type: 'string', required: true },
            { name: 'content', type: 'markdown' },
          ],
        },
      ]);
      expect(output).toEqualIgnoringWhitespace(`
        export interface Page {
          __metadata: { modelType: 'page'; id: string; urlPath: string; };
          type: 'Page';
          slug: string;
          content?: string;
        }
      `);
    });

    it('transforms complex model types', () => {
      expect(generateModelTypes(Object.values(models))).toMatchSnapshot();
    });
  });
});
