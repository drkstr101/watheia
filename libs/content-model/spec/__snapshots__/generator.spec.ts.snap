// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`watheia.cabbage/type-generator generateModelTypes transforms complex model types 1`] = `
"export interface BackgroundImage {
    __metadata: {
        modelType: 'object';
    };
    type: 'BackgroundImage';
    url?: string;
    backgroundSize: 'auto' | 'cover' | 'contain';
    backgroundPosition: 'bottom' | 'center' | 'left' | 'left-bottom' | 'left-top' | 'right' | 'right-bottom' | 'right-top' | 'top';
    backgroundRepeat: 'repeat' | 'repeat-x' | 'repeat-y' | 'no-repeat';
    opacity?: number;
}
export interface Button {
    __metadata: {
        modelType: 'object';
    };
    type: 'Button';
    label?: string;
    altText?: string;
    url: string;
    showIcon?: boolean;
    icon?: 'apple' | 'arrowLeft' | 'arrowLeftCircle' | 'arrowRight' | 'arrowRightCircle' | 'arrowUpLeft' | 'arrowUpRight' | 'cart' | 'chevronLeft' | 'chevronRight' | 'facebook' | 'github' | 'googlePlay' | 'instagram' | 'linkedin' | 'mail' | 'play' | 'playCircle' | 'reddit' | 'send' | 'twitter' | 'vimeo' | 'youtube';
    iconPosition?: 'left' | 'right';
    style: 'primary' | 'secondary';
    elementId?: string;
}
export interface CheckboxFormControl {
    __metadata: {
        modelType: 'object';
    };
    type: 'CheckboxFormControl';
    name: string;
    label?: string;
    width: 'full' | '1/2';
    isRequired?: boolean;
}
export interface Config {
    __metadata: {
        modelType: 'data';
        id: string;
    };
    type: 'Config';
    fixedLabel?: string;
    favicon?: string;
    header?: Header;
    footer?: Footer;
    titleSuffix?: string;
    defaultSocialImage?: string;
    defaultMetaTags?: (MetaTag)[];
}
export interface ContactBlock {
    __metadata: {
        modelType: 'object';
    };
    type: 'ContactBlock';
    title?: string;
    phoneNumber?: string;
    phoneAltText?: string;
    email?: string;
    emailAltText?: string;
    address?: string;
    addressAltText?: string;
    elementId?: string;
}
export interface ContactSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'ContactSection';
    title?: string;
    text?: string;
    form?: FormBlock;
    media?: ImageBlock | VideoBlock;
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    backgroundSize?: 'full' | 'inset';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface CtaSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'CtaSection';
    title?: string;
    text?: string;
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    backgroundSize?: 'full' | 'inset';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface DividerSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'DividerSection';
    title?: string;
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface EmailFormControl {
    __metadata: {
        modelType: 'object';
    };
    type: 'EmailFormControl';
    name?: string;
    label?: string;
    hideLabel?: boolean;
    placeholder?: string;
    width: 'full' | '1/2';
    isRequired?: boolean;
}
export interface FeaturedItem {
    __metadata: {
        modelType: 'object';
    };
    type: 'FeaturedItem';
    title?: string;
    subtitle?: string;
    text?: string;
    featuredImage?: ImageBlock;
    actions?: (Button | Link)[];
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface FeaturedItemsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'FeaturedItemsSection';
    title?: string;
    subtitle?: string;
    items?: (FeaturedItem)[];
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    columns?: number;
    spacingX?: number;
    spacingY?: number;
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface FeaturedPostsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'FeaturedPostsSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    posts?: (PostLayout)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    showDate?: boolean;
    showAuthor?: boolean;
    showExcerpt?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface FeaturedProjectsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'FeaturedProjectsSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    projects?: (ProjectLayout)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    showDate?: boolean;
    showDescription?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface Footer {
    __metadata: {
        modelType: 'object';
    };
    type: 'Footer';
    primaryLinks?: (Button | Link)[];
    contacts?: ContactBlock;
    copyrightText?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface FormBlock {
    __metadata: {
        modelType: 'object';
    };
    type: 'FormBlock';
    title?: string;
    fields?: (TextFormControl | EmailFormControl | TextareaFormControl | CheckboxFormControl | SelectFormControl)[];
    submitLabel?: string;
    elementId: string;
    action?: string;
    destination?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface Header {
    __metadata: {
        modelType: 'object';
    };
    type: 'Header';
    title?: string;
    isTitleVisible?: boolean;
    logo?: ImageBlock;
    primaryLinks?: (Link)[];
    socialLinks?: (Social)[];
    headerVariant: 'variant-a' | 'variant-b' | 'variant-c';
    isSticky?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface HeroSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'HeroSection';
    title?: string;
    subtitle?: string;
    text?: string;
    actions?: (Button | Link)[];
    media?: FormBlock | ImageBlock | VideoBlock;
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    backgroundSize?: 'full' | 'inset';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface ImageBlock {
    __metadata: {
        modelType: 'object';
    };
    type: 'ImageBlock';
    url?: string;
    altText?: string;
    caption?: string;
    elementId?: string;
}
export interface Label {
    __metadata: {
        modelType: 'object';
    };
    type: 'Label';
    label: string;
    url?: string;
}
export interface LabelsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'LabelsSection';
    title?: string;
    subtitle?: string;
    items?: (Label)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface Link {
    __metadata: {
        modelType: 'object';
    };
    type: 'Link';
    label?: string;
    altText?: string;
    url: string;
    showIcon?: boolean;
    icon?: 'apple' | 'arrowLeft' | 'arrowLeftCircle' | 'arrowRight' | 'arrowRightCircle' | 'arrowUpLeft' | 'arrowUpRight' | 'cart' | 'chevronLeft' | 'chevronRight' | 'facebook' | 'github' | 'googlePlay' | 'instagram' | 'linkedin' | 'mail' | 'play' | 'playCircle' | 'reddit' | 'send' | 'twitter' | 'vimeo' | 'youtube';
    iconPosition?: 'left' | 'right';
    elementId?: string;
}
export interface MediaGallerySection {
    __metadata: {
        modelType: 'object';
    };
    type: 'MediaGallerySection';
    title?: string;
    subtitle?: string;
    images?: (ImageBlock)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    spacing?: number;
    columns?: number;
    aspectRatio?: '1:1' | '3:2' | '2:3' | '4:3' | '3:4' | '16:9' | 'auto';
    showCaption?: boolean;
    enableHover?: boolean;
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface MetaTag {
    __metadata: {
        modelType: 'object';
    };
    type: 'MetaTag';
    property?: 'og:title' | 'og:type' | 'og:image' | 'og:image:alt' | 'og:url' | 'og:description' | 'og:locale' | 'og:site_name' | 'og:video' | 'twitter:card' | 'twitter:site' | 'twitter:creator' | 'twitter:description' | 'twitter:title' | 'twitter:image' | 'twitter:image:alt' | 'twitter:player';
    content?: string;
}
export interface PageLayout {
    __metadata: {
        modelType: 'page';
        id: string;
        urlPath: string;
    };
    type: 'PageLayout';
    title: string;
    sections?: ()[];
    metaTitle?: string;
    metaDescription?: string;
    addTitleSuffix?: boolean;
    socialImage?: string;
    metaTags?: (MetaTag)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e';
    backgroundImage?: BackgroundImage;
}
export interface Person {
    __metadata: {
        modelType: 'data';
        id: string;
    };
    type: 'Person';
    firstName: string;
    lastName?: string;
    role?: string;
    bio?: string;
    image?: ImageBlock;
}
export interface PostFeedLayout {
    __metadata: {
        modelType: 'page';
        id: string;
        urlPath: string;
    };
    type: 'PostFeedLayout';
    title?: string;
    postFeed?: PostFeedSection;
    topSections?: ()[];
    bottomSections?: ()[];
    metaTitle?: string;
    metaDescription?: string;
    addTitleSuffix?: boolean;
    socialImage?: string;
    metaTags?: (MetaTag)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e';
    backgroundImage?: BackgroundImage;
    styles?: Record<string, Record<string, unknown>>;
}
export interface PostFeedSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'PostFeedSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    showDate?: boolean;
    showAuthor?: boolean;
    showExcerpt?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface PostLayout {
    __metadata: {
        modelType: 'page';
        id: string;
        urlPath: string;
    };
    type: 'PostLayout';
    title: string;
    date: Date;
    author?: Person;
    excerpt?: string;
    featuredImage?: ImageBlock;
    media?: ImageBlock | VideoBlock;
    bottomSections?: ()[];
    metaTitle?: string;
    metaDescription?: string;
    addTitleSuffix?: boolean;
    socialImage?: string;
    metaTags?: (MetaTag)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e';
    backgroundImage?: BackgroundImage;
}
export interface ProjectFeedLayout {
    __metadata: {
        modelType: 'page';
        id: string;
        urlPath: string;
    };
    type: 'ProjectFeedLayout';
    title?: string;
    projectFeed?: ProjectFeedSection;
    topSections?: ()[];
    bottomSections?: ()[];
    metaTitle?: string;
    metaDescription?: string;
    addTitleSuffix?: boolean;
    socialImage?: string;
    metaTags?: (MetaTag)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e';
    backgroundImage?: BackgroundImage;
    styles?: Record<string, Record<string, unknown>>;
}
export interface ProjectFeedSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'ProjectFeedSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    showDate?: boolean;
    showDescription?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface ProjectLayout {
    __metadata: {
        modelType: 'page';
        id: string;
        urlPath: string;
    };
    type: 'ProjectLayout';
    title: string;
    date: Date;
    client?: string;
    description?: string;
    featuredImage?: ImageBlock;
    media?: ImageBlock | VideoBlock;
    bottomSections?: ()[];
    metaTitle?: string;
    metaDescription?: string;
    addTitleSuffix?: boolean;
    socialImage?: string;
    metaTags?: (MetaTag)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e';
    backgroundImage?: BackgroundImage;
}
export interface QuoteSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'QuoteSection';
    quote: string;
    name?: string;
    title?: string;
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface RecentPostsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'RecentPostsSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    recentCount?: number;
    showDate?: boolean;
    showAuthor?: boolean;
    showExcerpt?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface RecentProjectsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'RecentProjectsSection';
    title?: string;
    subtitle?: string;
    actions?: (Button | Link)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c' | 'variant-d';
    elementId?: string;
    recentCount?: number;
    showDate?: boolean;
    showDescription?: boolean;
    showFeaturedImage?: boolean;
    showReadMoreLink?: boolean;
    styles?: Record<string, Record<string, unknown>>;
}
export interface SelectFormControl {
    __metadata: {
        modelType: 'object';
    };
    type: 'SelectFormControl';
    name: string;
    label?: string;
    hideLabel?: boolean;
    defaultValue?: string;
    options?: string[];
    width: 'full' | '1/2';
    isRequired?: boolean;
}
export interface Social {
    __metadata: {
        modelType: 'object';
    };
    type: 'Social';
    label?: string;
    altText?: string;
    url: string;
    icon: 'facebook' | 'github' | 'instagram' | 'linkedin' | 'reddit' | 'twitter' | 'vimeo' | 'youtube';
    elementId?: string;
}
export interface Testimonial {
    __metadata: {
        modelType: 'object';
    };
    type: 'Testimonial';
    quote: string;
    name?: string;
    title?: string;
    image?: ImageBlock;
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface TestimonialsSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'TestimonialsSection';
    title?: string;
    subtitle?: string;
    testimonials?: (Testimonial)[];
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b' | 'variant-c';
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface TextareaFormControl {
    __metadata: {
        modelType: 'object';
    };
    type: 'TextareaFormControl';
    name: string;
    label?: string;
    hideLabel?: boolean;
    placeholder?: string;
    width: 'full' | '1/2';
    isRequired?: boolean;
}
export interface TextFormControl {
    __metadata: {
        modelType: 'object';
    };
    type: 'TextFormControl';
    name: string;
    label?: string;
    hideLabel?: boolean;
    placeholder?: string;
    width: 'full' | '1/2';
    isRequired?: boolean;
}
export interface TextSection {
    __metadata: {
        modelType: 'object';
    };
    type: 'TextSection';
    colors?: 'colors-a' | 'colors-b' | 'colors-c' | 'colors-d' | 'colors-e' | 'colors-f';
    variant: 'variant-a' | 'variant-b';
    title?: string;
    subtitle?: string;
    text?: string;
    elementId?: string;
    styles?: Record<string, Record<string, unknown>>;
}
export interface ThemeStyle {
    __metadata: {
        modelType: 'data';
        id: string;
    };
    type: 'ThemeStyle';
    white?: string | number;
    onLight?: string | number;
    black?: string | number;
    onDark?: string | number;
    primary?: string | number;
    onPrimary?: string | number;
    secondary?: string | number;
    onSecondary?: string | number;
    neutral?: string | number;
    onComplementary?: string | number;
    fontBody: 'fontBook' | 'fontDisplay';
    h1?: ThemeStyleHeading;
    h2?: ThemeStyleHeading;
    h3?: ThemeStyleHeading;
    h4?: ThemeStyleHeading;
    h5?: ThemeStyleHeading;
    h6?: ThemeStyleHeading;
    buttonPrimary?: ThemeStyleButton;
    buttonSecondary?: ThemeStyleButton;
    link?: ThemeStyleLink;
}
export interface ThemeStyleButton {
    __metadata: {
        modelType: 'object';
    };
    type: 'ThemeStyleButton';
    weight: 'normal' | 'medium';
    case: 'none' | 'lowercase' | 'capitalize' | 'uppercase';
    letterSpacing: 'tighter' | 'tight' | 'normal' | 'wide' | 'wider';
    horizontalPadding?: number;
    verticalPadding?: number;
}
export interface ThemeStyleHeading {
    __metadata: {
        modelType: 'object';
    };
    type: 'ThemeStyleHeading';
    size: 'xs' | 'sm' | 'base' | 'lg' | 'xl' | '2xl' | '3xl' | '4xl' | '5xl' | '6xl' | '7xl' | '8xl' | '9xl';
    weight: 'normal' | 'medium';
    decoration: 'none' | 'underline' | 'line-through';
    case: 'none' | 'lowercase' | 'capitalize' | 'uppercase';
    letterSpacing: 'tighter' | 'tight' | 'normal' | 'wide' | 'wider';
}
export interface ThemeStyleLink {
    __metadata: {
        modelType: 'object';
    };
    type: 'ThemeStyleLink';
    weight: 'normal' | 'medium';
    case: 'none' | 'lowercase' | 'capitalize' | 'uppercase';
    letterSpacing: 'tighter' | 'tight' | 'normal' | 'wide' | 'wider';
}
export interface VideoBlock {
    __metadata: {
        modelType: 'object';
    };
    type: 'VideoBlock';
    title?: string;
    url?: string;
    elementId?: string;
    autoplay?: boolean;
    loop?: boolean;
    muted?: boolean;
    controls?: boolean;
    aspectRatio?: '4:3' | '16:9';
}
"
`;
